### PR TITLE
Add knative-installer component to charts

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

Knative-installer is added to charts and installation scripts.

**Related issue(s)**

Uses #1748

### CODE REVIEW

**Comment by radufa** [comment]
It could be great if you can update the Windows related scripts (*.ps1) too?  They should remain convergent

File: installation/cmd/run.ps1 [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 $DOMAIN = "kyma.local"
+$MINIKUBE_EXTRA_ARGS = ""
+$CREATE_CR_EXTRA_ARGS = ""
+
+if ($KNATIVE -eq $false) {
+    $MINIKUBE_EXTRA_ARGS = "${MINIKUBE_ARGS} -memory 10240 -disk_size 30g"
+    $CREATE_CR_EXTRA_ARGS = "${CREATE_CR_EXTRA_ARGS} -crtpl_path ${CURRENT_DIR}/../resources/installer-cr-knative.yaml.tpl"
+}
+
+if ($SKIP_MINIKUBE_START -eq $false) {
```
Comment (crabtree) [inline installation/cmd/run.ps1]: This line should be removed
```suggestion
```

File: installation/cmd/run.ps1 [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
     [string]$VM_DRIVER = "hyperv"
 )
 
 $CURRENT_DIR = Split-Path $MyInvocation.MyCommand.Path
 $SCRIPTS_DIR = "${CURRENT_DIR}\..\scripts"
 $DOMAIN = "kyma.local"
+$MINIKUBE_EXTRA_ARGS = ""
+$CREATE_CR_EXTRA_ARGS = ""
+
+if ($KNATIVE -eq $false) {
```
Comment (crabtree) [inline installation/cmd/run.ps1]: This seems to work differently than in bash script. In `run.sh` you apply the knative cr when KNATIVE variable is non-zero, here you run this part when KNATIVE is false, which is default value for this flag.

File: installation/scripts/minikube.sh [hunk 3]
```diff
@@ -173,15 +185,16 @@ function start() {
...
-    --memory 8192 \
+    --memory $MEMORY \
     --cpus 4 \
     --extra-config=apiserver.Authorization.Mode=RBAC \
     --extra-config=apiserver.GenericServerRunOptions.CorsAllowedOriginList=".*" \
     --extra-config=controller-manager.ClusterSigningCertFile="/var/lib/localkube/certs/ca.crt" \
-	--extra-config=controller-manager.ClusterSigningKeyFile="/var/lib/localkube/certs/ca.key" \
-    --extra-config=apiserver.Admission.PluginNames="LimitRanger,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota" \
+    --extra-config=controller-manager.ClusterSigningKeyFile="/var/lib/localkube/certs/ca.key" \
+    --extra-config=apiserver.admission-control="LimitRanger,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota" \
```
Comment (crabtree) [inline installation/scripts/minikube.sh]: This line is inconsistent with PS script.

File: resources/knative-installer/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Knative installer
+
+## Overview
+
+This chart packs the [knative installation script](../../components/knative-installer/README.md) as a Kubernetes job.
+
+Installed releases of Knative components may be found in [Dockerfile](../../components/knative-installer/Dockerfile).
```
Comment (tomekpapiernik) [inline resources/knative-installer/README.md]:
```suggestion
For details on installed Knative components releases, see the [Dockerfile](../../components/knative-installer/Dockerfile).
```

File: installation/scripts/minikube.ps1 [hunk 5]
```diff
@@ -1,6 +1,8 @@
 param (
     [string]$VM_DRIVER = "hyperv",
     [string]$DOMAIN = "kyma.local"
+    [string]$DISK_SIZE = "20g"
```
Comment (radufa) [inline installation/scripts/minikube.ps1]: Please change here:
```
[string]$DOMAIN = "kyma.local",
[string]$DISK_SIZE = "20g",
```

**Comment by sjanota** [comment]
Image of knative-installer is broken. Fix in: #1776

**Pull request merged**